components/layout_2020/display_list/mod.rs

```diff
@@ -450,7 +451,67 @@ impl Fragment {
             self.build_display_list_for_text_decoration(fragment, builder, &rect, &color);
         }
 
-        // Text.
+        // TODO: This caret/text selection implementation currently does not account for vertical text.
+        if let Some(range) = fragment.selection_range {
+            let baseline_origin = rect.origin;
+            if !range.is_empty() {
+                let start = glyphs_advance_by_index(
+                    &fragment.glyphs,
+                    range.begin(),
+                    baseline_origin,
+                    fragment.justification_adjustment,
+                );
+
+                let end = glyphs_advance_by_index(
+                    &fragment.glyphs,
+                    range.end(),
+                    baseline_origin,
+                    fragment.justification_adjustment,
+                );
+
+                let selection_rect = LayoutRect::new(
```

Comment:
Does this work when start > end?

Reply:
It's not necessarily about RTL, but about reversed selection.

For example, if you create a selection by starting at the end and then keep pressing Shift+Left.

Reply:
https://github.com/servo/servo/pull/35830#issuecomment-2708213616

Here's a quick example that it works, again it's due to the Range retrieved is simply guaranteed to be start <= end.